cfgov/regulations3k/scripts/regml_importer.py

```diff
@@ -0,0 +1,427 @@
+        # Strip tags we're not interested in
+        etree.strip_tags(xml_tree, '{eregs}ref')
+        etree.strip_tags(xml_tree, '{eregs}def')
+        etree.strip_tags(xml_tree, '{eregs}callout')
+        etree.strip_tags(xml_tree, '{eregs}line')
+        etree.strip_tags(xml_tree, '{eregs}variable')
+        etree.strip_tags(xml_tree, '{eregs}subscript')
+
+        if part is None:
+            part = self.get_part(xml_tree)
+
+        if effective_version is None:
+            effective_version = self.get_effective_version(part, xml_tree)
+
+        subpart_nodes = xml_tree.findall(
+            './{eregs}part/{eregs}content/{eregs}subpart'
+        )
+        appendix_nodes = xml_tree.findall(
+            './{eregs}part/{eregs}content/{eregs}appendix'
+        )
+        interp_node = xml_tree.find(
+            './{eregs}part/{eregs}content/{eregs}interpretations'
+        )
+
+        if subpart_nodes is not None:
+            self.get_subparts(
+                effective_version, subpart_nodes, interp_node=interp_node
+            )
+
+        if appendix_nodes is not None:
+            self.get_appendices(
+                effective_version, appendix_nodes, interp_node=interp_node
+            )
+
+        if interp_node is not None:
+            self.get_interpretations(effective_version, interp_node)
+
+        return effective_version
+
+    def get_part(self, xml_tree):
+        part_number = xml_tree.find(
+            './{eregs}preamble/{eregs}cfr/{eregs}section'
+        ).text
+        cfr_title = xml_tree.find(
+            './{eregs}preamble/{eregs}cfr/{eregs}title'
+        ).text
+        part, created = Part.objects.get_or_create(
+            part_number=part_number,
+            chapter=CFR_CHAPTER,
+            cfr_title_number=cfr_title)
+
+        return part
+
+    def get_effective_version(self, part, xml_tree):
+        effective_date_str = xml_tree.find(
+            './{eregs}preamble/{eregs}effectiveDate'
+        ).text
+        effective_date = datetime.datetime.strptime(
+            effective_date_str, "%Y-%m-%d"
+        ).date()
+        effective_version, created = \
+            EffectiveVersion.objects.get_or_create(
+                effective_date=effective_date,
+                part=part
+            )
+
+        if not created:
+            raise ValueError(
+                'Effecitve date {} already exists for part {}'.format(
+                    effective_date, part)
+            )
+
+        effective_version.draft = True
+        return effective_version
+
+    def get_subparts(self, effective_version, subpart_nodes,
+                     interp_node=None):
+        subparts = []
+        for subpart_node in subpart_nodes:
+            label = subpart_node.get('label')
+            letter = subpart_node.get('subpartLetter')
+
+            title_text = ''
+            title_node = subpart_node.find('{eregs}title')
+            if title_node is not None:
```

Comment:
I think if title_node IS None, then we should set the title to "Sections" for the sake of regs such as B that have only one subpart for sections. This would match current content, rather than result in "Subpart None - " as a title.